core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java

```diff
@@ -43,6 +45,39 @@ public LogicalPlan visitAlias(Alias node, AnalysisContext context) {
     return node.getDelegated().accept(this, context);
   }
 
+  @Override
+  public LogicalPlan visitNestedAllTupleFields(NestedAllTupleFields node, AnalysisContext context) {
+    List<Map<String, ReferenceExpression>> args = new ArrayList<>();
+    for (NamedExpression namedExpr : namedExpressions) {
+
+      if (namedExpr.getDelegated() instanceof FunctionExpression
+          && ((FunctionExpression) namedExpr.getDelegated()).getFunctionName()
+          .getFunctionName().equalsIgnoreCase(BuiltinFunctionName.NESTED.name())) {
+
+        ReferenceExpression field =
+            (ReferenceExpression) ((FunctionExpression) namedExpr.getDelegated())
+                .getArguments().get(0);
+
+        // If path is same as NestedAllTupleFields path
+        if (field.getAttr().substring(0, field.getAttr().lastIndexOf("."))
+            .equalsIgnoreCase(node.getPath())) {
+          args.add(Map.of(
+              "field", field,
+              "path", new ReferenceExpression(node.getPath(), STRING)));
+        }
+      }
+    }
+
+    if (child instanceof LogicalNested) {
```

Comment:
What is `child`? It doesn't seem like a stateless analyzer. I feel it requires more comments (for `child` field and so on) and/or a doc.